src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1202,6 +1205,100 @@ private async Task<WebResponse> SendRequest(bool async)
+                        case HttpRequestCacheLevel.NoCacheNoStore:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.CacheControl.NoStore = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case HttpRequestCacheLevel.Reload:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case HttpRequestCacheLevel.CacheOnly:
+                        case HttpRequestCacheLevel.CacheOrNextCacheOnly:
+                            request.Headers.CacheControl.OnlyIfCached = true;
+                            break;
+                        case HttpRequestCacheLevel.Default:
+                            if (httpRequestCachePolicy.MinFresh > TimeSpan.Zero)
+                            {
+                                request.Headers.CacheControl.MinFresh = httpRequestCachePolicy.MinFresh;
+                            }
+
+                            if (httpRequestCachePolicy.MaxAge != TimeSpan.MaxValue)
+                            {
+                                request.Headers.CacheControl.MaxAge = httpRequestCachePolicy.MaxAge;
+                            }
+
+                            if (httpRequestCachePolicy.MaxStale > TimeSpan.Zero)
+                            {
+                                request.Headers.CacheControl.MaxStale = true;
+                                request.Headers.CacheControl.MaxStaleLimit = httpRequestCachePolicy.MaxStale;
+                            }
+
+                            break;
+                        case HttpRequestCacheLevel.Refresh:
+                            request.Headers.CacheControl.MaxAge = TimeSpan.Zero;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                    }
+                }
+                else
+                {
+                    switch (policy.Level)
+                    {
+                        case RequestCacheLevel.NoCacheNoStore:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.CacheControl.NoStore = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case RequestCacheLevel.Reload:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case RequestCacheLevel.CacheOnly:
+                            request.Headers.CacheControl.OnlyIfCached = true;
+                            break;
```

Comment:
`CacheOnly` is intended to throw if it's not in the local cache already. This is different from `OnlyIfCached`, which is a directive to the server/proxy.

.NET no longer has a local cache, so this can always throw a `WebException`. The text used in Framework is "The request was aborted: The request cache-only policy does not allow a network request and the response is not found in cache." and should be added as a resource string.